bin/cache/update_dart_sdk.sh

```diff
@@ -10,6 +10,12 @@ FLUTTER_ROOT=$(dirname $(dirname $(dirname "${BASH_SOURCE[0]}")))
 DART_SDK_PATH="$FLUTTER_ROOT/bin/cache/dart-sdk"
 DART_SDK_STAMP_PATH="$FLUTTER_ROOT/bin/cache/dart-sdk.stamp"
 DART_SDK_VERSION=`cat "$FLUTTER_ROOT/bin/cache/dart-sdk.version"`
+DART_CHANNEL="stable"
+
+if [[ $DART_SDK_VERSION == *"-dev."* ]]
+then
+  DART_CHANNEL="dev"
+fi
```

Comment:
We can do this work inside in the "if" below so that we only compute the channel when the version needs updating.